src/Compilers/CSharp/Portable/Compilation/MemberSemanticModel.cs

```diff
@@ -215,11 +221,16 @@ private Binder GetEnclosingBinder(CSharpSyntaxNode node, int position)
                         binder = RootBinder.GetBinder(current);
                     }
                 }
+                else if (current.Kind() == SyntaxKind.ArrowExpressionClause && current.Parent?.Kind() == SyntaxKind.LocalFunctionStatement)
```

Comment:
I don't think it will make code easier to understand. In fact, I believe it will make it much harder to understand.